### ISSUE TITLE
dtype equality operator does not check endianness of fields

### ISSUE THREAD

**Issue by bmerry** [numpy/numpy#5356]
For simple types, two dtype objects with different endianness are unequal:
```py
>>> np.dtype('<i4') == np.dtype('>i4')
False
```
However, for fields, the endianness is currently ignored when doing a comparison:
```py
>>> np.dtype([('a', '<i4')]) == np.dtype([('a', '>i4')])
True
```

This causes problems with other functions. For example,
```py
>>> big = np.dtype([('a', '>i2')])
>>> little = np.dtype([('a', '<i2')])
>>> arr = np.ones(4, dtype=big)
>>> print arr
[(1,) (1,) (1,) (1,)]
>>> print arr.astype(little)
[(256,) (256,) (256,) (256,)]
```
I'm guessing that astype thought the output dtype was the same, and hence did a memcpy instead of performing the endian conversion.

**Comment by jaimefrio** [comment]
This seems solved in current master. Can you confirm if it is still failing for you, @bmerry?

**Comment by bmerry** [comment]
It seems that it is fixed in master, and also in 1.9.2.

**Comment by jaimefrio** [comment]
See? If you wait long enough your problems go away on their own... ;-)

Thanks for rechecking.
